### PR TITLE
MOS-672: Summary page content

### PR DESCRIPTION
**Pull request by JAndresHJGlobant**
## What's included?
- Content component
- Storybook stories
- Unit tests
- Teal icon button variant

### CODE REVIEW

File: src/components/Content/Content.tsx [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+	const contentForDisplay = showOrHide ? content : content.slice(0, 3);
+
+	/**
+	 * Renders the content of type "labelValue"
+	 * @param contentItem 
+	 * @returns The content with a sctructure of label: value
+	 */
+	const labelValueContent = (contentItem: ContentType) => (
+			<LabelValueWrapper>
```
Comment (mauriciodeleonc) [inline src/components/Content/Content.tsx:46]: Codacy has a fix for the issue: [Expected indentation of 2 tabs but found 3.](https://app.codacy.com/gh/simpleviewinc/sv-mosaic/pullRequest?prid=9571838)
```suggestion
		<LabelValueWrapper>
```

File: src/components/Content/Content.tsx [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+				<Label>{contentItem.label}:</Label>
+				<Value>{contentItem.value}</Value>
```
Comment (mauriciodeleonc) [inline src/components/Content/Content.tsx:48]: Codacy has a fix for the issue: [Expected indentation of 3 tabs but found 4.](https://app.codacy.com/gh/simpleviewinc/sv-mosaic/pullRequest?prid=9571838)
```suggestion
			<Value>{contentItem.value}</Value>
```

File: src/components/Content/Content.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+export const Playground = (): ReactElement => {
+	const title = text("Title", "Main Section Title");
+	const showEdit = boolean("Show edit button", true);
+
+	return (
+		<Content
+			title={title}
+			content={content}
+			onClickEdit={showEdit && edit}
+			onClickAdd={onClickAdd}
```
Comment (mauriciodeleonc) [inline src/components/Content/Content.stories.tsx:70]: I think we should add a knob or something to make the onClickAdd actually do something, since it always has content, this will never be shown. Maybe a boolean knob that says "clear content" or "show as empty"

File: src/components/Content/Content.test.tsx [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+		expect(getAllByTestId("content-row").length).toBe(4);
+		expect(getByText("Fourth paragraph")).toBeDefined();
+	});
+});
+
+describe("Content componenent when no content is passed", () => {
+	beforeEach(() => {
+		render(
+			<Content title="Main Section" content={[]} onClickAdd={onClickAdd} />
```
Comment (mauriciodeleonc) [inline src/components/Content/Content.test.tsx:76]: I think we should add a test for when content = null or undefined because it's not required... I believe it should produce the same output as here, but just to be sure.

File: src/components/Content/Content.tsx [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+	ValueFile,
+	ContentRow,
+	ContentColumn,
+	ParagraphWrapper,
+} from "./Content.styled";
+import EditIcon from "@mui/icons-material/Edit";
+import AddIcon from "@mui/icons-material/Add";
+
+const Content = (props: ContentProps): ReactElement => {
+	const { title, onClickEdit, content, onClickAdd } = props;
```
Comment (mauriciodeleonc) [inline src/components/Content/Content.tsx:28]: Recommendation (you don't have to change this), I feel it looks / reads better "onEdit" and "onAdd" because "on" already talks about the action.

File: src/components/Content/Content.tsx [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+	const [showOrHide, setShowOrHide] = useState(content.length < 3);
```
Comment (mauriciodeleonc) [inline src/components/Content/Content.tsx:29]: Tip (you don't have to change this), you could leave the variable just as "show" since it's a boolean, show: false means that it's hidden (also I feel it's easier to read).

File: src/components/Content/Content.styled.tsx [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+	margin: 0;
+`;
+
+export const ParagraphWrapper = styled.div`
+	margin-bottom: 16px;
+`;
+
+// Styles for the LabelValue content type
+
+export const LabelValueWrapper = styled.div`
```
Comment (mauriciodeleonc) [inline src/components/Content/Content.styled.tsx:60]: Isn't this the same as the paragraph wrapper?

File: src/components/Content/Content.tsx [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+	};
+
+	const contentForDisplay = showOrHide ? content : content.slice(0, 3);
+
+	/**
+	 * Renders the content of type "labelValue"
+	 * @param contentItem 
+	 * @returns The content with a sctructure of label: value
+	 */
+	const labelValueContent = (contentItem: ContentType) => (
```
Comment (mauriciodeleonc) [inline src/components/Content/Content.tsx:45]: Isn't this the same as the paragraph content?

File: src/components/Content/Content.tsx [hunk 9]
```diff
@@ -0,0 +1,167 @@
...
+
+	/**
+	 * Renders the content of type "paragraph"
+	 * @param contentItem 
+	 * @returns A paragraph with the value of the content
+	 */
+	const paragraphContent = (contentItem: ContentType) => (
+		<ParagraphWrapper>
+			<Label>{contentItem.label}:</Label>
+			<Paragraph>{contentItem.value}</Paragraph>
```
Comment (mauriciodeleonc) [inline src/components/Content/Content.tsx:60]: I believe this is the only thing that changes between the labelValueContent and the paragraphContent, so you could reduce the repeated code by just creating a single element and passing a "type" prop to the styled component and that's where you decide what margin and font-size you apply.

File: src/components/Content/Content.tsx [hunk 10]
```diff
@@ -0,0 +1,167 @@
...
+	const fileContent = (contentItem: ContentType) => {
+		const FileIcon = contentItem?.icon;
+
+		return (
+			<>
+				<LabelWrapper>
+					{FileIcon && <FileIcon />}
+					<Label>{contentItem.label}:</Label>
+				</LabelWrapper>
+				<ValueFile>{contentItem.value}</ValueFile>
```
Comment (mauriciodeleonc) [inline src/components/Content/Content.tsx:101]: Same comment as with the paragraph.

File: src/components/Content/Content.tsx [hunk 11]
```diff
@@ -0,0 +1,167 @@
...
+		);
+	};
+
+	return (
+		<MainWrapper>
+			<TitleWrapper>
+				<Title>{title}</Title>
+				<ButtonsWrapper>
+					{content?.length === 0 ? (
+						<Button color="teal" variant="icon" mIcon={AddIcon} onClick={onClickAdd}></Button>
```
Comment (mauriciodeleonc) [inline src/components/Content/Content.tsx:112]: onClickAdd is optional, so this will still render even if the prop is not passed, which shouldn't happen.

File: src/components/Content/Content.tsx [hunk 12]
```diff
@@ -0,0 +1,167 @@
...
+						onClickEdit && (
+							<Button
+								color="gray"
+								variant="icon"
+								mIcon={EditIcon}
+								onClick={onClickEdit}
+							></Button>
+						)
+					)}
+					<Button
```
Comment (mauriciodeleonc) [inline src/components/Content/Content.tsx:123]: I believe this button will still render even if content.length <= 3

File: src/components/Content/Content.styled.tsx [hunk 13]
```diff
@@ -0,0 +1,107 @@
...
+	}
+`;
+
+export const ValueFile = styled(Value)`
+	font-size: 12px;
+	margin-left: 24px;
+`;
+
+export const ContentRow = styled.div`
+  display: flex;
```
Comment (mauriciodeleonc) [inline src/components/Content/Content.styled.tsx:101]: There are some weird tabs / spaces here and on width: 100%

File: src/components/Content/Content.tsx [hunk 14]
```diff
@@ -0,0 +1,167 @@
...
+					></Button>
+				</ButtonsWrapper>
+			</TitleWrapper>
+			<div>
+				{contentForDisplay.map((items, idx) => {
+					return (
+						<ContentRow data-testid={"content-row"} key={`${idx}-contentLayout`}>
+							{items.map((contentItem, idx) => {
+								if (
+									(contentItem.type === "tags" &&
```
Comment (mauriciodeleonc) [inline src/components/Content/Content.tsx:137]: All of this logic looks good, but I feel it might be better to process it outside of the render in something like a useEffect so we don't even render all the other components if this throws an error.

**Review by mauriciodeleonc — CHANGES_REQUESTED** [review]

File: src/components/Content/Content.styled.tsx (hunk 7 above)
Comment (JAndresHJGlobant) [reply src/components/Content/Content.styled.tsx:60]: You are right, I will use the same wrapper for both.

File: src/components/Content/Content.tsx (hunk 8 above)
Comment (JAndresHJGlobant) [reply src/components/Content/Content.tsx:45]: In this case, it is not. The ParagraphContent renders a `<p>` element whereas the LabelValue are `<span>` elements.

File: src/components/Content/Content.tsx (hunk 9 above)
Comment (JAndresHJGlobant) [reply src/components/Content/Content.tsx:60]: This cannot be done because of what I explained in a previous comment, they are different HTML elements.

File: src/components/Content/Content.tsx (hunk 10 above)
Comment (JAndresHJGlobant) [reply src/components/Content/Content.tsx:101]: This can be done since both are `<span>` I removed the ValueFile styled component and use the Value with a prop that defines the styles for each case.

File: src/components/Content/Content.tsx [hunk 15]
```diff
@@ -0,0 +1,196 @@
...
+const Content = (props: ContentProps): ReactElement => {
+	const { title, onEdit, content, onAdd } = props;
+	const [show, setShow] = useState(content?.length < 3);
+
+	/**
+	 * Validates if the combinations of the type of content and
+	 * value are allowed.
+	 */
+	useEffect(() => {
+		if(!content) return;
```
Comment (mauriciodeleonc) [inline src/components/Content/Content.tsx:41]: Codacy has a fix for the issue: [Expected space(s) after "if".](https://app.codacy.com/gh/simpleviewinc/sv-mosaic/pullRequest?prid=9571838)
```suggestion
		if (!content) return;
```

**Review by mauriciodeleonc — APPROVED** [review]
